force-app/main/default/lwc/nksSurvey/nksSurvey.js

```diff
@@ -0,0 +1,214 @@
+import { LightningElement, track, wire } from 'lwc';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import getSurvey from '@salesforce/apex/NKS_InternalSurveyController.getSurvey';
+import hasAnswered from '@salesforce/apex/NKS_InternalSurveyController.hasAnswered';
+import createAnsweredRecord from '@salesforce/apex/NKS_InternalSurveyController.createAnsweredRecord';
+import createResponseRecord from '@salesforce/apex/NKS_InternalSurveyController.createResponseRecord';
+export default class NksSurvey extends LightningElement {
+    surveyId;
+    backgroundColor;
+    title;
+    question;
+    isAnswered;
+    isRendered = false;
+    recordTypeId;
+    startDate;
+    endDate;
+
+    @track rating;
+    @track comment;
+    @track show = true;
+
+    @wire(getSurvey)
+    wiredSurvey({ data, error }) {
+        if (data) {
+            this.surveyId = data.Id;
+            this.backgroundColor = data.NKS_Background_Color__c;
+            this.title = data.NKS_Title__c;
+            this.question = data.NKS_Question__c;
+            this.startDate = data.NKS_Start_Date__c;
+            this.endDate = data.NKS_End_Date__c;
+        } else if (error) {
+            console.log('Problem getting survey: ' + error);
+        }
+    }
+
+    get background() {
+        return `background-color: ${this.backgroundColor}`;
+    }
+
+    get hide() {
+        return !this.show || this.isAnswered;
+    }
+
+    get isValid() {
+        let valid = false;
+        if (this.surveyId) {
+            if (this.startDate && this.endDate) {
+                valid = new Date(this.endDate) - new Date(this.startDate) > 0 ? true : false;
+            } else {
+                valid = true;
```

Comment:
Er survey valid hvis startdate eller enddate er null?

Reply:
Ja, det er satt opp sånt nå; tenker vi trenger tilbakemelding fra de som ska opprette surveys, for å avklare dette